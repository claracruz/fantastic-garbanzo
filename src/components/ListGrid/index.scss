.list-grid {
    list-style-type: none;
    padding: 0;
}
