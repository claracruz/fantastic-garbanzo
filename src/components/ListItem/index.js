import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';
import {faStarHalfAlt} from '@fortawesome/free-solid-svg-icons'
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

const months = [
	'Jan',
	'Feb',
	'Mar',
	'Apr',
	'May',
	'Jun',
	'Jul',
	'Aug',
	'Sep',
	'Oct',
	'Nov',
	"'Dec"
];


export const ListItem = (props) => {
	const {altmetric_id, details_url, images, published_on, score, title, url} = props.item;
	const datePublished = (published_on) ? new Date(published_on * 1000) : null;

	return (
		<li key={altmetric_id} className="list-item">
			<img src={images.small} alt={title}/>
			<span className="date-published">
				<span className="day">{datePublished && datePublished.getDay()}</span>
		        <span className="month">{months[datePublished && datePublished.getMonth()]}</span>
		        <span className="year">{datePublished && datePublished.getFullYear()}</span>
			</span>
			<div className="content">
				<h3>{title}</h3>

				<div className="details">
					<span className="score">
						<FontAwesomeIcon icon={faStarHalfAlt} />
						<span>{score}</span>
					</span>
					<a href={details_url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer" >View metrics data...</a>
					<a href={url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer">Go to source...</a>
				</div>
			</div>
		</li>
	);
};


ListItem.propTypes = {
	item: PropTypes.shape({
		altmetric_id: PropTypes.number,
		details_url: PropTypes.string,
		title: PropTypes.string,
		url: PropTypes.string,
		published_on: PropTypes.number,
		score: PropTypes.number,
		images: PropTypes.shape({
			small: PropTypes.string,
			medium: PropTypes.string,
			large: PropTypes.string
		})
	})
};
