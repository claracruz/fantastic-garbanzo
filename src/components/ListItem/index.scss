.list-item {

  background: #ffffff;
  height: auto;
  overflow: hidden;
  width: auto;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.3);
  margin: 15px 0;
  padding: 20px 20px 20px 75px;
  position: relative;
  display: flex;
  text-align: left;

  img {
    flex-shrink: 0;
    width: 64px;
    height: 64px;
  }

  .date-published {
    position: absolute;
    top: 0;
    left: 0;
    background-color: #77d7b9;
    color: #fff;
    padding: 0.8em;

    span {
      display: block;
      text-align: center;
      font-size: 0.8125rem;

      &.day {
        font-weight: 700;
        font-size: 24px;
        text-shadow: 2px 3px 2px rgba(0, 0, 0, 0.18);
      }

      &.month {
        text-transform: uppercase;
      }
    }
  }

  .content {
    margin: 0 25px;

    h3 {
      font-size: 1em;
      width: 75%;
      color: #444;
    }

    a {
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }

    .source {
      color: #0198A6;
      font-size: 0.8125rem;
      font-weight: bold;
      margin-left: 20px;
    }

    .details {
      margin-top: 15px;
      font-size: 0.8125rem;
      font-weight: bold;
      color: #0198A6;

      .score {

        span, a {
          position: relative;
          display: inline-block;
          left: 3px;
          top: 1px;
        }
      }
    }

  }
}
