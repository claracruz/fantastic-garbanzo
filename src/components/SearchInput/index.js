import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import './index.scss';

export class SearchInput extends PureComponent {

	constructor(props) {
		super(props);
		this.state = {
			searchTerm: ''
		}
	}

	onSubmit = () => {
		this.props.onSearch(this.state.searchTerm);
	};

	handleChange = (e) => {
		this.setState({
			searchTerm: e.currentTarget.value
		});
	};

	handleKeyDown = (e) => {
		if (e.key === 'Enter') {
			this.onSubmit();
		}
	};

	render() {
		return (
			<div className="wrap">
				<h2><label htmlFor="search">Search Papers</label></h2>
				<div className="search">
					<input
						id="search"
						className="search-input"
						value={this.state.searchTerm}
						onKeyPress={this.handleChange}
						onKeyDown={this.handleKeyDown}
						onChange={this.handleChange}
						placeholder="Enter a search term"
						type="text" />
					<button className="searchButton" onClick={this.onSubmit}><FontAwesomeIcon icon={faSearch} /></button>
				</div>
			</div>
		);
	}
}

SearchInput.propTypes = {
	onSearch: PropTypes.func.isRequired
};
