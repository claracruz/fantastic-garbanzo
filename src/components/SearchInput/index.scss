
.wrap {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 20px;
    height: 65px;

    h2 {
        margin-right: 40px;
        position: relative;
        top: 3px;
    }

    .search {
        max-width: 320px;
        width: 100%;
        position: relative;
        margin-right: 35px;
    }

    .search-input {;
        width: 100%;
        border: 3px solid #0198A6;
        padding: 5px;
        height: 36px;
        border-radius: 5px;
        outline: none;
        font-size: 0.8125rem;
        background-color: white;
    }

    .searchButton {
        position: absolute;
        right: -32px;
        width: 40px;
        height: 36px;
        border: 1px solid #0198A6;
        background: #0198A6;
        text-align: center;
        color: #fff;
        border-top-right-radius: 5px;
        border-bottom-right-radius: 5px;
        cursor: pointer;
        font-size: 1rem;
    }
}


