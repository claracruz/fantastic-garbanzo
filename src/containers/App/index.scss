.App {

  transition: all 0.5s ease;

  .App-header {
    width: 100%;
    height: 55px;
    background-color: #0198A6;
    color: white;
    padding: 12px;
    font-size: 2rem;
    display: flex;

    h1 {
      padding: 0;
      margin: 0 10px;
      font-size: 1.5rem;
    }

    svg {

    }
  }

  nav {
    display: block;
    overflow: hidden;
    background-color: white;
    padding-bottom: 15px;
    //height: 78px;
  }

  main {
    height: 100%;
    width: 100%;
    text-align: center;
    padding: 20px;

    h2 {
      margin: 0 0 20px 0;
      font-size: 1.2rem;
    }

    .paginationInfo {
      float: right;
      color: #777;
      font-size: 0.75rem;
    }

    .pagination {
      list-style-type: none;

      li {
        display: inline-block;
      }

      button {
        color: black;
        float: left;
        padding: 8px 16px;
        text-decoration: none;
        transition: background-color .3s;
        border: 1px solid #ddd;
        margin: 0 2px;
        font-size: 0.8125rem;
        cursor: pointer;

        &.active {
          background-color: #0198A6;
          color: white;
          border: 1px solid #0198A6;
        }

        &:hover:not(.active) {background-color: #f3f3f3;}
      }
    }
  }

  .error {
    margin: 1px auto 0 20px;
    color: red;
    font-size: 0.875em;
  }

  .list-grid {
    width: 100%;
    margin: 0 auto;
  }
}
