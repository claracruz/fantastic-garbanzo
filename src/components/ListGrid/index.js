import React  from 'react';
import PropTypes from 'prop-types';
import { ListItem } from '../ListItem';
import './index.scss';


function displayData (props) {
	const { items } = props;
	return items.map((item) => {
		return (
			<ListItem key={item.altmetric_id} item={item}/>
		);
	})
}

export const ListGrid = (props) => {
	const { className, items } = props;
	const hasData = items.length > 0;

	return (
		<>
			{
				hasData &&
				<ul className={className}>{displayData(props)}</ul>
			}
		</>
	);
};

ListGrid.propTypes = {
	items: PropTypes.arrayOf(
		PropTypes.shape({
			altmetric_id: PropTypes.number,
			title: PropTypes.string,
			url: PropTypes.string,
			details_url: PropTypes.string,
			published_on: PropTypes.number,
			score: PropTypes.number,
			images: PropTypes.shape({
				small: PropTypes.string,
				medium: PropTypes.string,
				large: PropTypes.string
			}),
		})
	)
};

ListGrid.defaultProps = {
	items: []
};
