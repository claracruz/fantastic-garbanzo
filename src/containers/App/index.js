import React, { PureComponent } from 'react';
import { ListGrid } from '../../components/ListGrid/index';
import { SearchInput } from '../../components/SearchInput/index';
import { Loading } from '../../components/Loading/index';
import services  from '../../utils/requests';
import './index.scss';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faAtlas} from "@fortawesome/free-solid-svg-icons";
import classnames from 'classnames';

class App extends PureComponent {

	constructor(props) {
		super(props);
		this.state = {
			error: '',
			loading: false,
			queryData: null,
			searchResults: [],
			searchTerm: ''
		}
	}

	onSearch = async (value) => {
		if (value) {
			this.setState({ loading: true });
			const data = await services.fetchData(value);
			this.onDataFetched(data, value);
		}
	};

	onDataFetched(data, value) {
		let newState = {
			loading: false
		};
		if (data.error) {
			Object.assign(newState, {
				error: `"${value}" ${data.error}`,
				queryData: null,
				searchResults: [],
				searchTerm: ''
			})
		} else {
			Object.assign(newState, {
				error: '',
				queryData: data.query,
				searchResults: data.results,
				searchTerm: value
			});
		}
		this.setState(newState);
	}

	showPaginationInfo(numberOfPages, currentPage) {
		const pageNodes = new Array(numberOfPages).fill(null);
		return <ul className="pagination">
			{
				pageNodes.map((node, index) => {
					const pageIndex = index + 1;
					return (
						<li key={`page_${pageIndex}`}>
							<button
								className={classnames({
									active: pageIndex === currentPage
								})}
								onClick={this.onNavigate}>{pageIndex}</button>
						</li>
					);
				})
			}
		</ul>
	}

	onNavigate() {
		//fetch pagination data
		alert('not implemented!')
	}

	render() {
		const { error, loading, queryData, searchResults, searchTerm } = this.state;
		const numPages = (queryData) ? Math.ceil(queryData.num_results/queryData.total) : null;

		return (
			<div className="App">
				<header className="App-header">
					<FontAwesomeIcon icon={faAtlas} />
					<h1>Search App</h1>
				</header>
				<nav>
					<SearchInput onSearch={this.onSearch} />
					{ error && <div className="error">{error}</div> }
				</nav>
				{ !!queryData &&
					<main>
						<span className="paginationInfo">
							<span>{`Page ${queryData.page}`}</span>
							<span> of </span>
							<span>{numPages}</span>
						</span>
						<h2>{`Showing ${searchResults.length} results of ${queryData.num_results} for "${searchTerm}"`}</h2>
						<ListGrid
							className="list-grid"
							items={searchResults} />
						{this.showPaginationInfo(numPages, queryData.page)}
					</main>
				}
				{ loading && <Loading /> }
			</div>
		);
	}
}

export default App;
