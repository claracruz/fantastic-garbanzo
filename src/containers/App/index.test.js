import React from 'react';
import App  from './index';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import services  from '../../utils/requests';
import { ListGrid } from '../../components/ListGrid';
import { Loading } from '../../components/Loading';
import enzymeSerializer from 'enzyme-to-json/serializer';

expect.addSnapshotSerializer(enzymeSerializer);

describe('<App />', () => {

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders correctly', () => {
		const wrapper = shallow(<App />);
		const searchInput = wrapper.find('SearchInput');
		const h1 = wrapper.find('h1');
		expect(wrapper).toMatchSnapshot();
		expect(searchInput.length).toBe(1);
		expect(h1.length).toBe(1);
	});

	describe('State handling', () => {
		it ('renders with default state', () => {
			const wrapper = shallow(<App />);
			expect(wrapper.instance().state).toEqual({
				error: '',
				loading: false,
				queryData: null,
				searchResults: [],
				searchTerm: ''
			});
		});

		it('renders as expected when loading state occurs', () => {
			const wrapper = shallow(<App />);
			wrapper.instance().setState({loading: true});
			expect(wrapper).toMatchSnapshot();

			const loading = <Loading />;
			expect(wrapper.contains(loading)).toEqual(true);
		});

		it('renders as expected when state has data', () => {
			const wrapper = shallow(<App />);
			const queryData = {total: 12, page: 1, num_results: 25};
			const searchResults = [
				{ altmetric_id: 122, title: 'title1', url: 'http://url/1', details_url: 'http://detailsUrl/1'},
				{ altmetric_id: 234, title: 'title2', url: 'http://url/2', details_url: 'http://detailsUrl/1'},
				{ altmetric_id: 543, title: 'title3', url: 'http://url/3', details_url: 'http://detailsUrl/1'}
			];
			const searchTerm = 'test term';

			wrapper.instance().setState({searchResults, searchTerm, queryData});
			expect(wrapper).toMatchSnapshot();

			const listGrid = wrapper.find('ListGrid');
			expect(listGrid.props().items).toEqual(searchResults);

			const paginationInfo = wrapper.find('.paginationInfo');
			expect(paginationInfo.length).toBe(1);
			expect(paginationInfo.text()).toEqual('Page 1 of 3');

			const resultsInfo = wrapper.find('h2');
			expect(resultsInfo.length).toBe(1);
			expect(resultsInfo.text()).toEqual(`Showing 3 results of 25 for "${searchTerm}"`);
		});

		it('renders as expected when error state occurs', () => {
			const wrapper = shallow(<App />);
			const error = 'This is a test error';

			wrapper.instance().setState({error});
			expect(wrapper).toMatchSnapshot();

			const errorNode = wrapper.find('.error');
			expect(errorNode.length).toBe(1);
			expect(errorNode.text()).toEqual(error);
		});
	});

	describe('onSearch()', () => {
		it('handles onSearch() as expected when no search term passed', () => {
			const wrapper = shallow(<App />);

			services.fetchData = jest.fn(() => Promise.resolve({ }));
			wrapper.instance().onSearch();
			expect(services.fetchData.mock.calls.length).toBe(0);
			expect(wrapper).toMatchSnapshot();
		});

		it('handles onSearch() as expected when data returned', async () => {
			const wrapper = shallow(<App />);
			const searchTerm = 'test search term';
			const queryData = {total: 12, page: 1, num_results: 25};
			const searchResults = [
				{ altmetric_id: 122, title: 'title1', url: 'http://url/1', details_url: 'http://detailsUrl/1'},
				{ altmetric_id: 234, title: 'title2', url: 'http://url/2', details_url: 'http://detailsUrl/1'},
				{ altmetric_id: 543, title: 'title3', url: 'http://url/3', details_url: 'http://detailsUrl/1'}
			];

			wrapper.instance().setState = jest.fn();
			services.fetchData = jest.fn(() => Promise.resolve({ query: queryData, results: searchResults }));
			await wrapper.instance().onSearch(searchTerm);

			expect(services.fetchData.mock.calls.length).toBe(1);
			expect(services.fetchData.mock.calls[0][0]).toEqual(searchTerm);

			const setStateMock = wrapper.instance().setState;
			expect(setStateMock.mock.calls.length).toBe(2);
			expect(setStateMock.mock.calls[0][0]).toEqual({ loading: true });
			expect(setStateMock.mock.calls[1][0]).toEqual({
				loading: false,
				error: '',
				queryData,
				searchResults,
				searchTerm
			});

			expect(wrapper).toMatchSnapshot();


		});

		it('handles onSearch() as expected when error response received', async () => {
			const wrapper = shallow(<App />);
			const searchTerm = 'test search term';
			const error = 'this is a test error response';

			wrapper.instance().setState = jest.fn();
			services.fetchData = jest.fn(() => Promise.resolve({ error }));
			await wrapper.instance().onSearch(searchTerm);

			expect(services.fetchData.mock.calls.length).toBe(1);
			expect(services.fetchData.mock.calls[0][0]).toEqual(searchTerm);

			const setStateMock = wrapper.instance().setState;
			expect(setStateMock.mock.calls.length).toBe(2);
			expect(setStateMock.mock.calls[0][0]).toEqual({ loading: true });
			expect(setStateMock.mock.calls[1][0]).toEqual({
				error: `"${searchTerm}" ${error}`,
				loading: false,
				queryData: null,
				searchResults: [],
				searchTerm: ""
			});

			expect(wrapper).toMatchSnapshot();
		});
	});
});
