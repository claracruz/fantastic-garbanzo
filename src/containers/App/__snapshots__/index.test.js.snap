// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> State handling renders as expected when error state occurs 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
    <div
      className="error"
    >
      This is a test error
    </div>
  </nav>
</div>
`;

exports[`<App /> State handling renders as expected when loading state occurs 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
  <Loading />
</div>
`;

exports[`<App /> State handling renders as expected when state has data 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
  <main>
    <span
      className="paginationInfo"
    >
      <span>
        Page 1
      </span>
      <span>
         of 
      </span>
      <span>
        3
      </span>
    </span>
    <h2>
      Showing 3 results of 25 for "test term"
    </h2>
    <ListGrid
      className="list-grid"
      items={
        Array [
          Object {
            "altmetric_id": 122,
            "details_url": "http://detailsUrl/1",
            "title": "title1",
            "url": "http://url/1",
          },
          Object {
            "altmetric_id": 234,
            "details_url": "http://detailsUrl/1",
            "title": "title2",
            "url": "http://url/2",
          },
          Object {
            "altmetric_id": 543,
            "details_url": "http://detailsUrl/1",
            "title": "title3",
            "url": "http://url/3",
          },
        ]
      }
    />
    <ul
      className="pagination"
    >
      <li
        key="page_1"
      >
        <button
          className="active"
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        key="page_2"
      >
        <button
          className=""
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        key="page_3"
      >
        <button
          className=""
          onClick={[Function]}
        >
          3
        </button>
      </li>
    </ul>
  </main>
</div>
`;

exports[`<App /> onSearch() handles onSearch() as expected when data returned 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
</div>
`;

exports[`<App /> onSearch() handles onSearch() as expected when error response received 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
</div>
`;

exports[`<App /> onSearch() handles onSearch() as expected when no search term passed 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
</div>
`;

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f558",
            "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
          ],
          "iconName": "atlas",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h1>
      Search App
    </h1>
  </header>
  <nav>
    <SearchInput
      onSearch={[Function]}
    />
  </nav>
</div>
`;
